qemu/tests/in_place_upgrade.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import aexpect
+
+
+from virttest import error_context
+from provider.in_place_upgrade_base import IpuTest
+
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Run in place upgrade cases:
+    a) without RHSM
+    1.configure vm
+    2.install leapp tool
+    3.download new rhel content repo
+    4.pre_upgrade test in the vm
+    5.upgrade test in the vm
+    6.check if it's target system
+
+    b) with rhsm
+    1.configure vm
+    2.install leapp tool
+    3.subscribe vm
+    4.pre_upgrade test in the vm
+    5.upgrade test in the vm
+    6.check if it's target system
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    upgrade_test = IpuTest(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    upgrade_test.session = vm.wait_for_login(timeout=login_timeout)
+    try:
+        # set post_release
+        pre_release = params.get("pre_release")
+        release_chk = params.get("release_check")
+        if pre_release not in upgrade_test.run_guest_cmd(release_chk):
+            test.cancel("your image is not for rhel 8 product, please check")
+        post_release = params.get("post_release")
+        # create an assistant user
+        upgrade_test.create_ipuser(test)
+        # prepare ipu test env and execute leapp tool
+        upgrade_test.run_guest_cmd(params.get("repo_leapp"))
+        upgrade_test.run_guest_cmd(params.get("ins_leapp_cmd"))
+        upgrade_test.run_guest_cmd(params.get("prepare_env"))
+        upgrade_test.run_guest_cmd(params.get("get_answer_files_source"))
+        upgrade_test.run_guest_cmd(params.get("get_answer_files"))
+        if params.get("rhsm_type") == "no_rhsm":
+            # update vm
+            old_custom_repo = params.get("old_custom_internal_repo")
```

Comment:
Because those links are our company internal repo. Considering security, I put them in the internal cfg file. But I will give comments to explain it more, is it ok ? Thanks

Reply:
Should be commented/explain in the cfg file, not only in the GitHub comment.

Reply:
will update it, thanks.